nodejs/eks/nodegroup.ts

```diff
@@ -1750,18 +1744,18 @@ function createManagedNodeGroupInternal(
         delete nodeGroupArgs.diskSize;
     }
 
+    let amiType = args.amiType;
+
     // amiType, releaseVersion and version cannot be set if an AMI ID is set in a custom launch template.
     // The AMI ID is set in the launch template if custom user data is required.
     // See https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html#mng-ami-id-conditions
     if (requiresCustomUserData(customUserDataArgs)) {
         delete nodeGroupArgs.version;
         delete nodeGroupArgs.releaseVersion;
         delete nodeGroupArgs.amiType;
-    }
+    } else if (amiType === undefined && args.operatingSystem !== undefined) {
+        // if no ami type is provided, but operating system is provided, determine the ami type based on the operating system
```

Comment:
👍 these narrative comments quickly fall out of date, but looks like this is still fine, it's in the "else" branch because it can actually set ami in this case.